managed-configuration-chrome.py

```diff
@@ -0,0 +1,118 @@
+configuration.per_page_limit = 20
+# int | The initial index from which to return the results.(optional) default to 0)
+configuration.per_page_offset = 0
+
+
+
+# get all the groups present in the enterprise and execute run_managed_configuration on each of them
+def get_all_groups_in_enterprise():
+    # create an instance of the API class
+    allgroups = []
+    api_instance = esperclient.DeviceGroupApi(esperclient.ApiClient(configuration))
+    try:
+        api_response = api_instance.get_all_groups(enterprise_id,
+                                                    limit=configuration.per_page_limit,
+                                                    offset=configuration.per_page_offset)
+        if len(api_response.results):
+            for group in api_response.results:
+                # add all the devices in this group to global list of devices
+                allgroups.append(group.id)
+                run_managed_configuration(allgroups)
+        #print(allgroups)
+
+    except ApiException as e:
+        print("Exception when calling DeviceGroupApi->get_all_groups: %s\n" % e)
+
+# V0CommandRequest | The request body to create a command for set of devices or groups
+## command_type ->
+##  * DEVICE: command request is meant for devices
+##  * GROUP: command request is meant for groups
+##  * DYNAMIC: command request is meant for dynamic set of devices i.e subset of devices from different groups or otherwise.
+## Type of devices to run commands on
+##  * active: Run commands on currently online devices
+##  * inactive: Run commands on currently offline devices
+##  * all: Run commands on all the devices. Commands will be queued for offline devices until they come back online.
+
+# run this on each group one by one
+def run_managed_configuration(allgroups):
+
+    request = esperclient.V0CommandRequest(
+        enterprise=enterprise_id,
+        command_type="GROUP",
+        device_type="all",
+        groups = allgroups,
+        command="UPDATE_DEVICE_CONFIG",
+        command_args=command_args
+    )
+
+    try:
+        # Create a command request
+        api_response = api_instance.create_command(enterprise_id, request)
+        #print(api_response)
+
+        request_id = api_response.id
+        response = api_instance.get_command_request_status(enterprise_id, request_id)
+        status = response.results[0]
+        #print(status)
+
+        while status.state not in ["Command Success", "Command Failure", "Command TimeOut", "Command Cancelled", "Command Queued"]:
+            response = api_instance.get_command_request_status(enterprise_id, request_id)
+            status = response.results[0]
+            time.sleep(1)
+            #print(status)
+
+    except ApiException as e:
+        print("Exception when calling CommandsV2Api->create_command: %s\n" % e)
+
+
+"""
+Main Function
+"""
+def main():
+    get_all_groups_in_enterprise()
+    run_managed_configuration()
```

Comment:
Function prototype is has modified. Also you're doing all operations inside 'get_all_groups_in_enterprise' function, so this function call is not needed I guess.

Reply:
Updated. Thanks